cfgov/v1/models/base.py

```diff
@@ -181,12 +181,19 @@ def specific_categories_query(block, parent_slug):
                     block, 'archive-past-events') & query
             relate = block.value.get('relate_{}'.format(search_type), None)
             if relate:
-                related[search_type_name] = (
+                type_query = (
                     AbstractFilterPage.objects.live().filter(
                         search_query
                     ).distinct().exclude(id=self.id).order_by(
                         '-date_published'
-                    )[:block.value['limit']])
+                    )
+                )
+                # Apply similar logic as snippets.py's filter_by_tags method
+                # to enable AND filtering
+                if block.value['and_filtering']:
+                    for tag in self.tags.names() or []:
```

Comment:
Is there a reason you have `or []` here?  If there are no tags, it will return an empty array.